lightning/src/routing/network_graph.rs

```diff
@@ -264,18 +264,7 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
 	/// does not match our chain_hash will be rejected when the announcement is
 	/// processed.
 	fn handle_reply_channel_range(&self, their_node_id: &PublicKey, msg: ReplyChannelRange) -> Result<(), LightningError> {
-		log_debug!(self.logger, "Handling reply_channel_range peer={}, first_blocknum={}, number_of_blocks={}, full_information={}, scids={}", log_pubkey!(their_node_id), msg.first_blocknum, msg.number_of_blocks, msg.full_information, msg.short_channel_ids.len(),);
-
-		// Validate that the remote node maintains up-to-date channel
-		// information for chain_hash. Some nodes use the full_information
-		// flag to indicate multi-part messages so we must check whether
-		// we received SCIDs as well.
-		if !msg.full_information && msg.short_channel_ids.len() == 0 {
-			return Err(LightningError {
-				err: String::from("Received reply_channel_range with no information available"),
-				action: ErrorAction::IgnoreError,
-			});
-		}
+		log_debug!(self.logger, "Handling reply_channel_range peer={}, first_blocknum={}, number_of_blocks={}, sync_complete={}, scids={}", log_pubkey!(their_node_id), msg.first_blocknum, msg.number_of_blocks, msg.sync_complete, msg.short_channel_ids.len(),);
```

Comment:
Okay if I follow correctly, the fact we're stateless means we can't qualify if the range is effectively complete and `sync_complete` sets properly. But we don't care as peer was able to equivocate before, logging is best we can do.

Reply:
Right - we don't care if a given message indicates the end or not - we just respond to them as they come in.